### PR TITLE
[incubator/jaeger] Add security context, trim README

### PR DESCRIPTION
**Pull request by naseemkullah**
Signed-off-by: Naseem <naseemkullah@gmail.com>

#### What this PR does / why we need it:
* Add security context
* Trim README
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: incubator/jaeger/templates/agent-ds.yaml [hunk 1]
```diff
@@ -37,60 +37,62 @@ spec:
...
       dnsPolicy: {{ .Values.agent.dnsPolicy }}
-      nodeSelector:
-{{ toYaml .Values.agent.nodeSelector | indent 8 }}
-{{- if .Values.agent.tolerations }}
-      tolerations:
-{{ toYaml .Values.agent.tolerations | indent 8 }}
-{{- end }}
       serviceAccountName: {{ template "jaeger.agent.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.agent.podSecurityContext | nindent 8 }}
```
Comment (pavelnikolov) [inline incubator/jaeger/templates/agent-ds.yaml:42]: Why do you use `nindent` instead of `ident` here? What is the reason for the empty line at the beginning?

File: incubator/jaeger/templates/agent-ds.yaml [hunk 2]
```diff
@@ -37,60 +37,62 @@ spec:
...
+              port: admin
+          readinessProbe:
+            httpGet:
+              path: /
+              port: admin
+          resources:
+            {{ toYaml .Values.agent.resources | indent 12 }}
+      nodeSelector:
+        {{ toYaml .Values.agent.nodeSelector | indent 8 }}
+      tolerations:
```
Comment (pavelnikolov) [inline incubator/jaeger/templates/agent-ds.yaml:96]: Node select and tolerations should be sent as a separate PR. They do not seem to be related to security context, right?

File: incubator/jaeger/templates/cassandra-schema-job.yaml [hunk 3]
```diff
@@ -25,21 +25,25 @@ spec:
...
-          value: {{ .Values.schema.mode | quote }}
-        - name: DATACENTER
-          value: {{ .Values.cassandra.config.dc_name | quote }}
-        - name: CASSANDRA_PORT
-          value: {{ .Values.storage.cassandra.port | quote }}
-        resources:
-{{ toYaml .Values.schema.resources | indent 10 }}
+        - name: {{ include "jaeger.fullname" . }}-cassandra-schema
+          securityContext:
+              {{- toYaml .Values.schema.securityContext | nindent 12 }}
```
Comment (pavelnikolov) [inline incubator/jaeger/templates/cassandra-schema-job.yaml:33]: Again, why do you need to use `nindent`?

File: incubator/jaeger/templates/collector-deploy.yaml [hunk 4]
```diff
@@ -43,69 +43,73 @@ spec:
...
+          readinessProbe:
+            httpGet:
+              path: /
+              port: admin
+          livenessProbe:
+            httpGet:
+              path: /
+              port: admin
+          resources:
+            {{ toYaml .Values.collector.resources | indent 12 }}
```
Comment (pavelnikolov) [inline incubator/jaeger/templates/collector-deploy.yaml:112]: Why was this indentation increased to 12? Am I missing something? Please, revert.

File: incubator/jaeger/templates/spark-cronjob.yaml [hunk 5]
```diff
@@ -29,40 +29,42 @@ spec:
...
+                      key: password
+                - name: ES_USERNAME
+                  value: {{ .Values.storage.elasticsearch.user }}
+                {{- end }}
+              resources:
+                {{ toYaml .Values.spark.resources | indent 16 }}
+          nodeSelector:
+            {{ toYaml .Values.spark.nodeSelector | indent 12 }}
+          tolerations:
+            {{ toYaml .Values.spark.tolerations | indent 12 }}
```
Comment (pavelnikolov) [inline incubator/jaeger/templates/spark-cronjob.yaml:68]: Is node selector and tolerations needed for the pod security context? If not, please, send as a separate PR.

File: incubator/jaeger/templates/spark-cronjob.yaml [hunk 6]
```diff
@@ -29,40 +29,42 @@ spec:
...
+                - name: ES_PASSWORD
+                  valueFrom:
+                    secretKeyRef:
+                      name: {{ if .Values.storage.elasticsearch.existingSecret }}{{ .Values.storage.elasticsearch.existingSecret }}{{- else }}{{ include "jaeger.fullname" . }}-elasticsearch{{- end }}
+                      key: password
+                - name: ES_USERNAME
+                  value: {{ .Values.storage.elasticsearch.user }}
+                {{- end }}
+              resources:
+                {{ toYaml .Values.spark.resources | indent 16 }}
```
Comment (pavelnikolov) [inline incubator/jaeger/templates/spark-cronjob.yaml:64]: Why is everything here indented? This is unrelated change, please, revert.

File: incubator/jaeger/templates/query-deploy.yaml [hunk 7]
```diff
@@ -36,89 +36,92 @@ spec:
...
+            containerPort: 14271
+            protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /
+              port: admin
+          readinessProbe:
+            httpGet:
+              path: /
+              port: admin
```
Comment (pavelnikolov) [inline incubator/jaeger/templates/query-deploy.yaml:120]: This whole indentation is unrelated change. Please, revert it.

File: incubator/jaeger/templates/collector-deploy.yaml (hunk 4 above)
Comment (pavelnikolov) [reply incubator/jaeger/templates/collector-deploy.yaml:112]: The indentation change is unrelated to security context. Please, revert it and keep one logical change per PR.

**Review by pavelnikolov — CHANGES_REQUESTED** [review]
Please, send one logical change per PR for easier reviews. Ideally, this should be split into 3 different PRs:
- add security context + README.md changes 
- nodeSelector + tolerations
- indentation and formatting changes

File: incubator/jaeger/templates/agent-ds.yaml (hunk 1 above)
Comment (naseemkullah) [reply incubator/jaeger/templates/agent-ds.yaml:42]: Good point.
I added this as boilerplate for helm create command via https://github.com/helm/helm/pull/5919/files I will make subsequent request to remove the unneeded new line! And adjust it here as well.

Comment (naseemkullah) [reply incubator/jaeger/templates/agent-ds.yaml:42]: Oh wait no I think I get it. Due to the `-` in  ``{{- toYaml`` which will remove all empty space, without the newline it sill begin pasting directly after `securityContext:`

File: incubator/jaeger/templates/agent-ds.yaml (hunk 2 above)
Comment (naseemkullah) [reply incubator/jaeger/templates/agent-ds.yaml:96]: That is correct and you are right about making 1 PR per logical change. or at the very least 1 git commit per logical change. I do apologize and if you would be so kind as to just let this one slide I'll make sure to stop this bad habit moving forward. Seeing that you've already reviewed everything! :p

File: incubator/jaeger/templates/cassandra-schema-job.yaml (hunk 3 above)
Comment (naseemkullah) [reply incubator/jaeger/templates/cassandra-schema-job.yaml:33]: My understanding is that since we have a `-` all whitespace will be removed, so the newline will help if not we would get something like:
```
          securityContext:            capabilities:
              drop:
              - ALL
            readOnlyRootFilesystem: true
            runAsNonRoot: true
            runAsUser: 10001
```

File: incubator/jaeger/templates/spark-cronjob.yaml (hunk 6 above)
Comment (naseemkullah) [reply incubator/jaeger/templates/spark-cronjob.yaml:64]: My intention is that charts be somewhat standardized. The helm create command uses an indentation for lists. I actually preferred no indentation and made a PR https://github.com/helm/helm/pull/5725 ... However after getting feedback from at least 2 helm maintainers and some of my colleagues, the general consensus is that indented lists are more readable.

File: incubator/jaeger/templates/cassandra-schema-job.yaml (hunk 3 above)
Comment (naseemkullah) [reply incubator/jaeger/templates/cassandra-schema-job.yaml:33]: Due to the `-` in `{{- toYaml`` which removes previous whitespace

File: incubator/jaeger/templates/collector-deploy.yaml (hunk 4 above)
Comment (naseemkullah) [reply incubator/jaeger/templates/collector-deploy.yaml:112]: I will wait to see if you let this one slide... 1 logical change per PR is definitely the way to go and I will strive for this moving forward. Just seeing that you took the time to go over the changes, maybe we could let this one slide :)

File: incubator/jaeger/templates/spark-cronjob.yaml (hunk 5 above)
Comment (naseemkullah) [reply incubator/jaeger/templates/spark-cronjob.yaml:68]: I will wait to see if you let this one slide... 1 logical change per PR is definitely the way to go and I will strive for this moving forward. Just seeing that you took the time to go over the changes, maybe we could let this one slide :)

File: incubator/jaeger/templates/query-deploy.yaml (hunk 7 above)
Comment (naseemkullah) [reply incubator/jaeger/templates/query-deploy.yaml:120]: My intention is that charts be somewhat standardized. The helm create command uses an indentation for lists. I actually preferred no indentation and made a PR helm/helm#5725 ... However after getting feedback from at least 2 helm maintainers and some of my colleagues, the general consensus is that indented lists are more readable.

Comment (naseemkullah) [reply incubator/jaeger/templates/query-deploy.yaml:120]: But it is indeed unrelated to security context

File: incubator/jaeger/templates/spark-cronjob.yaml (hunk 6 above)
Comment (pavelnikolov) [reply incubator/jaeger/templates/spark-cronjob.yaml:64]: Yes, I agree 100% that indentation is easier to read. However, it makes this particular PR harder to reader/review. I'd still prefer a separate PR with just the indentation and then rebasing this branch to reflect the changes.

File: incubator/jaeger/templates/spark-cronjob.yaml (hunk 5 above)
Comment (pavelnikolov) [reply incubator/jaeger/templates/spark-cronjob.yaml:68]: I really appreciate what you've done in this PR and I would be happy to merge this in. But indeed, I'd prefer 3 separate PRs. I promise to review them quickly.

**Comment by naseemkullah** [comment]
What a shame

**Comment by pavelnikolov** [comment]
Thank you for your understanding. Let me know if you need any help with splitting this.

**Comment by naseemkullah** [comment]
Thanks. I believe I will need to start from scratch again
